src/toolbars/ToolBarButtons.cpp

```diff
@@ -0,0 +1,125 @@
+#include <utility>
+#include <wx/app.h>
+
+#include "ToolBarButtons.h"
+
+#include "AllThemeResources.h"
+#include "../widgets/AButton.h"
+#include "Project.h"
+
+#include "../commands/CommandContext.h"
+#include "../commands/CommandManager.h"
+#include "../commands/CommandDispatch.h"
+
+// flags so 1,2,4,8 etc.
+enum {
+   TBActTooltips = 1,
+   TBActEnableDisable = 2,
+};
+
+ToolBarButtons::ToolBarButtons(ToolBar *const parent, AudacityProject & project, ButtonList buttonList, int size, int firstButtonId)
+   : mParent(parent)
+   , mProject(project)
+   , mButtonList(std::move(buttonList))
+   , mFirstButtonId(firstButtonId)
+{
+   mButtons.resize(size);
```

Comment:
I prefer member initializations where possible instead of mutations inside the constructor body.

But beware, it should be
`mButtons(size)`

not the curly-bracket form!  Because `vector` has constructors taking `initializer_list` and you don't want overload resolution to one of those.